src/firewall/core/rich.py

```diff
@@ -661,10 +674,16 @@ def check(self):
             if self.element.protocol not in [ "tcp", "udp", "sctp", "dccp" ]:
                 raise FirewallError(errors.INVALID_PROTOCOL, self.element.protocol)
 
-        # other element and not empty?
-        elif self.element is not None:
-            raise FirewallError(errors.INVALID_RULE, "Unknown element %s" % 
-                                type(self.element))
+        # tcp-mss-clamp
+        elif type(self.element) == Rich_Tcp_Mss_Clamp:
+            if self.element.value:
+                if not functions.checkTcpMssClamp(self.element.value):
+                    raise FirewallError(errors.INVALID_RULE, self.element.value)
```

Comment:
We also need to check that no action is specified. `tcp-mss-clamp` is an implicit action. This is similar to `masquerade`. This will also cause a change in your expected test output for nftables rule listings.
See here:

https://github.com/firewalld/firewalld/pull/678/files#diff-12066b4f26399116ab8d7b0c4ebc6198R629